truss/remote/baseten/auth.py

```diff
@@ -0,0 +1,54 @@
+import functools
+import logging
+import os
+from typing import Optional
+
+from truss.remote.baseten.error import AuthorizationError
+
+logger = logging.getLogger(__name__)
+
+
+class ApiKey:
+    value: str
+
+    def __init__(self, value: str):
+        self.value = value
+
+    def headers(self):
+        return {"Authorization": f"Api-Key {self.value}"}
+
+
+class AuthService:
+    def __init__(self, api_key: Optional[str] = None):
+        if api_key is not None:
+            self.set_key(api_key)
+
+    def validate(self):
```

Comment:
why do we need to mutate the environment in this method? It does not look like we read from this env var anywhere else. It looks frankly from this that we should need a way of setting global state, and this is not the cleanest way of achieving this.

Nor do I think you need to do this. In `BasetenRemote`, you are already threading the api key through to the api module.

Reply:
We do read from this env var from the CLI. + for posterity, we want to enforce auth keys as env vars (at least for this implementation). We may move away from this implementation but for now, auth keys are encoded in the env.